pkg/repositories/transformers/filters.go

```diff
@@ -86,6 +90,62 @@ func constructModelFilter(ctx context.Context, singleFilter *datacatalog.SingleP
 				JoinCondition: gormimpl.NewGormJoinCondition(sourceEntity, common.Tag),
 			}, nil
 		}
+	case *datacatalog.SinglePropertyFilter_DatasetFilter:
+		switch datasetProperty := propertyFilter.DatasetFilter.GetProperty().(type) {
+		case *datacatalog.DatasetPropertyFilter_Project:
+			project := datasetProperty.Project
+			logger.Debugf(ctx, "Constructing Dataset filter project:[%v]", project)
+			if err := validators.ValidateEmptyStringField(datasetProperty.Project, "project"); err != nil {
+				return models.ModelFilter{}, err
+			}
+			projectFilter := gormimpl.NewGormValueFilter(operator, projectFieldName, project)
+			modelValueFilters := []models.ModelValueFilter{projectFilter}
+
+			return models.ModelFilter{
```

Comment:
just a nit: to reduce some of the repetition here since you return 
```
return models.ModelFilter{
    Entity:       common.Dataset,
    ValueFilters: modelValueFilters,
}, nil
``` 
in every case, you could declare the variables outside the switch statement and assign them based on the filter property

Reply:
👍 thanks